packages/melos/lib/src/commands/version.dart

```diff
@@ -439,10 +441,23 @@ Hint: try running "melos version --all" to include private packages.
 
     final pubspec = File(pubspecPathForDirectory(Directory(package.path)));
     final contents = await pubspec.readAsString();
-    final updatedContents = contents.replaceAllMapped(
-        dependencyVersionReplaceRegex(dependencyName), (Match match) {
-      return '${match.group(1)}$dependencyVersion';
-    });
+    String updatedContents;
+
+    final gitReference =
+        package.pubSpec.dependencies[dependencyName] is GitReference ||
+            package.pubSpec.devDependencies[dependencyName] is GitReference;
+
+    if (gitReference) {
```

Comment:
We should probably only do this if `--git-tag-version` is true right?

Reply:
That makes sense yes.
I will check how we do that and update

Reply:
Okay @Salakar 
I double checked this flag. (The documentation needs updating as it is just a copy/paste from `--no-changelog`) 

However. It seems like this option still updates the versions in `pubspec.yaml` only skips the git commit (which obviously also does not tag).

If this is indeed the intended behaviour (skipping the commit) then I think we should still do the update, but do introduce the cofig in `melos.yaml`